proto/gossip.proto

```diff
@@ -44,6 +46,61 @@ message GossipRequest {
 message GossipResponse {
   // Requested delta of server's infostore.
   optional bytes delta = 1;
+  // TODO(thschroeter): replace bytes with GossipInfoStore
+  //optional GossipInfoStore delta = 1;
   // Non-nil means client should retry with this address.
   optional Addr alternate = 2;
 }
+
+message GossipInfoStore {
+  // Map from key to info
+  map<string, GossipInfo> infos = 1;
```

Comment:
so this is interesting. @thschroeter does this actually work? maps are a protobuf 3 feature, and this file (and all the others) are explicitly protobuf 2 syntax.

I'm ok with introducing some protobuf 3, especially if it helps achieve specific goals (like getting us off of gob), but we should do it piecemeal and separately from existing protos. In this case, we'd want to put this proto in a new package.